- Recently I got a lot of feedback from my dear friends who just change or about the change their career towards to Data Analysis/ Data Science and Machine Learning areas about the lack of material between beginning the analysis journey and the advanced techniques.

- They are looking for detailed but at the same time beginner friendly, not so much complicated (with different regression, normalization techniques, etc.) explained Explanatory Data Analysis examples, which show them how to start and most importantly how to read the descriptive statistics and graphs.

- After getting these feedbacks, I have decided to make some kind of series of EDA’s from different datasets, without making so complicated for the people at their first steps of DS/ML journey.

### This notebook is part of the 9 Beginner Friendly EDAs. If these EDAs would be helpful to anyone, I would be more than happy.



#### **INTRO**



- In this study, we are going to make Exploratory Data Analysis (EDA) with the Hollywood's Most Profitable Movies dataset. 
- Study aims to be beginner friendly and give as much as possible explanation for each step on the way.
- Study's dataset has 74 movies along with their ratings, profitability, worldwide gross and leading studio.
- Data includes 2007-2011 movies.

- First, let's import the required libraries.
- We will use Plotly's interactive environment for visualization.

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt 
import seaborn as sns 
import matplotlib as mpl


import plotly 
import plotly.express as px
import plotly.graph_objs as go
import plotly.offline as py
from plotly.offline import iplot
from plotly.subplots import make_subplots
import plotly.figure_factory as ff

### Overview Stage

- Read the csv
- Look for basic information about the dataset

In [2]:
df= pd.read_csv('../input/hollywood-most-profitable-stories/HollywoodsMostProfitableStories.csv')

In [3]:
df.head()

Film    Genre  Lead Studio  Audience  score %  \
0            27 Dresses   Comedy          Fox               71.0   
1  (500) Days of Summer   Comedy          Fox               81.0   
2    A Dangerous Method    Drama  Independent               89.0   
3         A Serious Man    Drama    Universal               64.0   
4   Across the Universe  Romance  Independent               84.0   

   Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
0       5.343622               40.0       160.308654  2008  
1       8.096000               87.0        60.720000  2009  
2       0.448645               79.0         8.972895  2011  
3       4.382857               89.0        30.680000  2009  
4       0.652603               54.0        29.367143  2007

In [4]:
df.shape

(74, 8)

- We have 74 films and 8 variables to work on

In [5]:
df.isnull().sum()

Film                 0
Genre                0
Lead Studio          1
Audience  score %    1
Profitability        3
Rotten Tomatoes %    1
Worldwide Gross      0
Year                 0
dtype: int64

- We have several missing values, which we need to look into.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74 entries, 0 to 73
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Film               74 non-null     object 
 1   Genre              74 non-null     object 
 2   Lead Studio        73 non-null     object 
 3   Audience  score %  73 non-null     float64
 4   Profitability      71 non-null     float64
 5   Rotten Tomatoes %  73 non-null     float64
 6   Worldwide Gross    74 non-null     float64
 7   Year               74 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.8+ KB


- As a data type, everything is in order to work on.

In [7]:
df.describe()

Audience  score %  Profitability  Rotten Tomatoes %  Worldwide Gross  \
count          73.000000      71.000000          73.000000        74.000000   
mean           64.136986       4.741610          47.356164       136.351979   
std            13.647665       8.292017          26.242655       157.067561   
min            35.000000       0.005000           3.000000         0.025000   
25%            52.000000       1.790680          27.000000        32.447500   
50%            64.000000       2.642353          45.000000        73.198612   
75%            76.000000       4.850958          65.000000       190.185250   
max            89.000000      66.934000          96.000000       709.820000   

              Year  
count    74.000000  
mean   2009.054054  
std       1.353756  
min    2007.000000  
25%    2008.000000  
50%    2009.000000  
75%    2010.000000  
max    2011.000000

Before going further, let's summarize what we have got from the dataset.

- Our dataset has 74 films from different genres and lead studios.

- Object data type variables, like genre and lead studio can be grouped and see the differences among them.

- There are several  missing values to look for it. 

-  Numerical variables deserves special attention for further analysis.


- Let's make the necessary adjustments before moving to the analysis part.

#### Missing Values

- Let's remember, which columns have the missing values

In [8]:
df.isnull().sum()

Film                 0
Genre                0
Lead Studio          1
Audience  score %    1
Profitability        3
Rotten Tomatoes %    1
Worldwide Gross      0
Year                 0
dtype: int64

- Since we will use all the films, film column must be without missing values. So we are OK for the film column.
- Genre, Worldwide Gross and Year columns do not have any missing values. Hurray!!
- Let's see the missing values.

In [9]:
df[df['Lead Studio'].isnull()]

Film   Genre Lead Studio  Audience  score %  Profitability  \
38  No Reservations  Comedy         NaN               64.0        3.30718   

    Rotten Tomatoes %  Worldwide Gross  Year  
38               39.0         92.60105  2007

- In addition to domain knowledge and expertise, there are tons of different ways to deal with the missing values.

- Most of the time, people tend to use drop function or use fillna function to use that row in their analysis. As I said, it depends on the data, domain knowledge and importance of the variable in our analysis.

- For our analysis in this dataset, main variable which we want to work on it, is 'Film'. If we have the name of the film and also values for at least some of the other variables, we can use that row.

- Based on aferomentioned points, we can keep this row, it has a lot of usefull information.


In [10]:
df[df['Audience  score %'].isnull()]

Film    Genre  Lead Studio  Audience  score %  \
50  Something Borrowed  Romance  Independent                NaN   

    Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
50       1.719514                NaN           60.183  2011

- Audience score and Rotten Tomates scores are good variables to use for rating purposes. 
- But in this row we don't have any of them.
- But still we have values for other columns, so we can keep this row.

In [11]:
df[df['Profitability'].isnull()]

Film    Genre  Lead Studio  Audience  score %  \
18           Jane Eyre  Romance    Universal               77.0   
41  Our Family Wedding   Comedy  Independent               49.0   
70        When in Rome   Comedy       Disney               44.0   

    Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
18            NaN               85.0           30.147  2011  
41            NaN               14.0           21.370  2010  
70            NaN               15.0           43.040  2010

- Profitability has 3 missing values.
- Even though we don't have profitability values in these rows, we have values for other columns.
- So better to keep them.

#### Look at the 'Genre' and 'Lead Studio'

In [12]:
df['Genre'].value_counts()

Comedy       41
Romance      15
Drama        13
Animation     3
Action        1
Fantasy       1
Name: Genre, dtype: int64

- Seems quite OK to use in the groupby.
- Noted

In [13]:
df['Lead Studio'].value_counts()

Independent              19
Warner Bros.             12
Universal                 7
Disney                    7
Fox                       6
Summit                    5
Sony                      4
Paramount                 4
The Weinstein Company     3
20th Century Fox          2
Lionsgate                 2
CBS                       1
New Line                  1
Name: Lead Studio, dtype: int64

- It can be used.
- OK it is not in a perfect shape to use in group by, but not bad at all to use. 
- Noted.

- Everything seems OK.  Let's move on to the next step: **analysis part**.

### Analysis Part

#### Genre

In [14]:
df['Genre'].value_counts(normalize=True)

Comedy       0.554054
Romance      0.202703
Drama        0.175676
Animation    0.040541
Action       0.013514
Fantasy      0.013514
Name: Genre, dtype: float64

- 55% of the movies are in the Comedy genre
- Romance and Drama follows it 

In [15]:
fig = px.histogram(df, x="Genre", title='Genre')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

#### Lead Studio

In [16]:
df['Lead Studio'].value_counts(normalize=True)

Independent              0.260274
Warner Bros.             0.164384
Universal                0.095890
Disney                   0.095890
Fox                      0.082192
Summit                   0.068493
Sony                     0.054795
Paramount                0.054795
The Weinstein Company    0.041096
20th Century Fox         0.027397
Lionsgate                0.027397
CBS                      0.013699
New Line                 0.013699
Name: Lead Studio, dtype: float64

- Independent studios make 26% of the all the studios in this dataset.
- Warner Bros also comes close by 16% 

In [17]:
fig = px.histogram(df, x="Lead Studio", title='Lead Studios')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

### Audience Score %

In [18]:
df['Audience  score %'].describe()

count    73.000000
mean     64.136986
std      13.647665
min      35.000000
25%      52.000000
50%      64.000000
75%      76.000000
max      89.000000
Name: Audience  score %, dtype: float64

- Based on the mean and median values, auidence score seems quite normally distributed.
- Both mean and median value is 64.

In [19]:

fig = px.histogram(df, x= 'Audience  score %', title='Percentage of the Audience  score', marginal="box", hover_data = df[['Film','Genre']])


fig.show()

#### Profitability

In [20]:
df['Profitability'].describe()

count    71.000000
mean      4.741610
std       8.292017
min       0.005000
25%       1.790680
50%       2.642353
75%       4.850958
max      66.934000
Name: Profitability, dtype: float64

- We have rightly skewed distribution (mean is signicantly bigger than median)
- Which basicaly means, we have possible outliers in our dataset and they affect mean value.
- Let's see it.

In [21]:
fig = px.histogram(df, x= 'Profitability', title='Profitability of the Movies', marginal="box", hover_data = df[['Film','Genre']])


fig.show()

- Yeah, we have a quite a number of outliers and rightly skewed distribution of the Profitability of the movies.

#### Rotten Tomatoes %

In [22]:
df['Rotten Tomatoes %'].describe()

count    73.000000
mean     47.356164
std      26.242655
min       3.000000
25%      27.000000
50%      45.000000
75%      65.000000
max      96.000000
Name: Rotten Tomatoes %, dtype: float64

- We have maximum number of 96 in the data. It affects mean score.
- We can expect rightly skewed distribution but not that much extend.
- Because mean and median scores are close to each others (47 & 45)

In [23]:
fig = px.histogram(df, x= 'Rotten Tomatoes %', title='Rating Score-Rotten Tomatoes %', marginal="box", hover_data = df[['Film','Genre']])


fig.show()

In [24]:
df['Worldwide Gross'].describe()

count     74.000000
mean     136.351979
std      157.067561
min        0.025000
25%       32.447500
50%       73.198612
75%      190.185250
max      709.820000
Name: Worldwide Gross, dtype: float64

- Yep, as you correctly see, we have highly rightly skewed distribution (mean= 136.5, median = 73.19)
- We have possible outliers.
- Let's see it

In [25]:
fig = px.histogram(df, x= 'Worldwide Gross', title='Worldwide Gross', marginal="box", hover_data = df[['Film','Genre']])
fig.show()

- Befor moving on the details, let's see the correlation matrix for our dataset

In [26]:
df.drop('Year', axis=1).corr()

Audience  score %  Profitability  Rotten Tomatoes %  \
Audience  score %           1.000000       0.042083           0.601990   
Profitability               0.042083       1.000000           0.024210   
Rotten Tomatoes %           0.601990       0.024210           1.000000   
Worldwide Gross             0.395357       0.146705           0.019748   

                   Worldwide Gross  
Audience  score %         0.395357  
Profitability             0.146705  
Rotten Tomatoes %         0.019748  
Worldwide Gross           1.000000

In [27]:
index_vals = df['Genre'].astype('category').cat.codes

fig = go.Figure(data=go.Splom(
                dimensions=[dict(label='Audience  score %',
                                 values=df['Audience  score %']),
                            dict(label='Profitability',
                                 values=df['Profitability']),
                            dict(label='Rotten Tomatoes %',
                                 values=df['Rotten Tomatoes %']),
                            dict(label='Worldwide Gross',
                                 values=df['Worldwide Gross'])],
                showupperhalf=False, 
                text=df['Film'],
                marker=dict(color=index_vals,
                            showscale=False, # colors encode categorical variables
                            line_color='white', line_width=0.5)
                ))


fig.update_layout(
    title='Movies',
    width=1000,
    height=1000,
)

fig.show()

Based on the results:
- There is positive but not so strong relationship (.60) between Audience Score and Rotten Tomatoes
- Also there is positive but weak (.395) relationship between Audience Score and the Worldwide Gross.

- After getting overall picture about the data, we can go into more details.

In [28]:
genre_by_year = df.groupby('Year')['Genre'].value_counts().reset_index(level=0).rename(columns={'Genre': 'Genre count'}, index={'index': 'Genre'})
genre_by_year

Year  Genre count
Genre                       
Comedy     2007            6
Romance    2007            5
Comedy     2008           12
Drama      2008            3
Romance    2008            2
Animation  2008            1
Fantasy    2008            1
Comedy     2009            7
Drama      2009            5
Comedy     2010           15
Drama      2010            2
Action     2010            1
Animation  2010            1
Romance    2011            8
Drama      2011            3
Animation  2011            1
Comedy     2011            1

In [29]:
fig = px.line(genre_by_year, x='Year', y='Genre count', color= genre_by_year.index, title='Movies By Genre in Each Year')
fig.show()

- From the line plot we can see that Movies in the Comedy and Drama genres do not have consistency by year
- Movies in Romance genre, increased significantly after 2008 
- Animation movies are stable in count by year. 

### Profitability by Year

In [30]:
fig = px.scatter(df, x='Year', y='Profitability', title='Movies By Profitability in Each Year')
fig.show()

### Worldwide Gross by Year

In [31]:
fig = px.scatter(df, x='Year', y='Worldwide Gross', title='Movies By Worldwide Gross in Each Year')
fig.show()

- Let's look at the top 15 WorldWide Gross Movies

#### Top 15 WorldWide Gross Movies

In [32]:
top_15 = df.sort_values('Worldwide Gross', ascending=False)[:15]
top_15

Film      Genre   Lead Studio  \
59          The Twilight Saga: New Moon      Drama        Summit   
62              Twilight: Breaking Dawn    Romance   Independent   
29                           Mamma Mia!     Comedy     Universal   
67                               WALL-E  Animation        Disney   
47                     Sex and the City     Comedy  Warner Bros.   
61                             Twilight    Romance        Summit   
51                              Tangled  Animation        Disney   
7                             Enchanted     Comedy        Disney   
57                         The Proposal     Comedy        Disney   
48                   Sex and the City 2     Comedy  Warner Bros.   
53  The Curious Case of Benjamin Button    Fantasy  Warner Bros.   
15   High School Musical 3: Senior Year     Comedy        Disney   
17                     It's Complicated     Comedy     Universal   
69                What Happens in Vegas     Comedy           Fox   
21                           Knocked Up     Comedy     Universal   

    Audience  score %  Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
59               78.0      14.196400               27.0       709.820000  2009  
62               68.0       6.383364               26.0       702.170000  2011  
29               76.0       9.234454               53.0       609.473955  2008  
67               89.0       2.896019               96.0       521.283432  2008  
47               81.0       7.221796               49.0       415.253258  2008  
61               82.0      10.180027               49.0       376.661000  2008  
51               88.0       1.365692               89.0       355.080000  2010  
7                80.0       4.005737               93.0       340.487652  2007  
57               74.0       7.867500               43.0       314.700000  2009  
48               49.0       2.883500               15.0       288.350000  2010  
53               81.0       1.783944               73.0       285.431000  2008  
15               76.0      22.913136               65.0       252.044501  2008  
17               63.0       2.642353               56.0       224.600000  2009  
69               72.0       6.267647               28.0       219.367646  2008  
21               83.0       6.636402               91.0       219.001261  2007

In [33]:
fig = px.bar(top_15, x='Film', y= 'Worldwide Gross',  hover_data = top_15[['Year','Genre']], color='Genre')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

- Let's see top 15 Rotten Tomatoes % Rating Movies

#### Top 15 Rotten Tomatoes % Rating Movies

In [34]:
rotten_tomatoes_top_15 = df.sort_values('Rotten Tomatoes %', ascending=False)[:15]
rotten_tomatoes_top_15

Film      Genre            Lead Studio  \
67                              WALL-E  Animation                 Disney   
31                   Midnight in Paris    Romance                   Sony   
7                            Enchanted     Comedy                 Disney   
21                          Knocked Up     Comedy              Universal   
51                             Tangled  Animation                 Disney   
3                        A Serious Man      Drama              Universal   
66                            Waitress    Romance            Independent   
1                 (500) Days of Summer     Comedy                    Fox   
45              Rachel Getting Married      Drama            Independent   
18                           Jane Eyre    Romance              Universal   
5                            Beginners     Comedy            Independent   
35                My Week with Marilyn      Drama  The Weinstein Company   
2                   A Dangerous Method      Drama            Independent   
32      Miss Pettigrew Lives for a Day     Comedy            Independent   
37  Nick and Norah's Infinite Playlist     Comedy                   Sony   

    Audience  score %  Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
67               89.0       2.896019               96.0       521.283432  2008  
31               84.0       8.744706               93.0       148.660000  2011  
7                80.0       4.005737               93.0       340.487652  2007  
21               83.0       6.636402               91.0       219.001261  2007  
51               88.0       1.365692               89.0       355.080000  2010  
3                64.0       4.382857               89.0        30.680000  2009  
66               67.0      11.089742               89.0        22.179483  2007  
1                81.0       8.096000               87.0        60.720000  2009  
45               61.0       1.384167               85.0        16.610000  2008  
18               77.0            NaN               85.0        30.147000  2011  
5                80.0       4.471875               84.0        14.310000  2011  
35               84.0       0.825800               83.0         8.258000  2011  
2                89.0       0.448645               79.0         8.972895  2011  
32               70.0       0.252895               78.0        15.173694  2008  
37               67.0       3.352729               73.0        33.527293  2008

In [35]:
fig = px.bar(rotten_tomatoes_top_15, x='Film', y= 'Rotten Tomatoes %',  hover_data = rotten_tomatoes_top_15[['Year','Genre']], color='Genre')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

- Let's see top 15 Audience score % Rating Movies

#### Top 15 Audience score % Rating Movies

In [36]:
audience_score_top_15 = df.sort_values('Audience  score %', ascending=False)[:15]
audience_score_top_15

Film      Genre            Lead Studio  \
2                    A Dangerous Method      Drama            Independent   
67                               WALL-E  Animation                 Disney   
51                              Tangled  Animation                 Disney   
31                    Midnight in Paris    Romance                   Sony   
4                   Across the Universe    Romance            Independent   
35                 My Week with Marilyn      Drama  The Weinstein Company   
21                           Knocked Up     Comedy              Universal   
61                             Twilight    Romance                 Summit   
43                      P.S. I Love You    Romance            Independent   
53  The Curious Case of Benjamin Button    Fantasy           Warner Bros.   
1                  (500) Days of Summer     Comedy                    Fox   
47                     Sex and the City     Comedy           Warner Bros.   
5                             Beginners     Comedy            Independent   
7                             Enchanted     Comedy                 Disney   
59          The Twilight Saga: New Moon      Drama                 Summit   

    Audience  score %  Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
2                89.0       0.448645               79.0         8.972895  2011  
67               89.0       2.896019               96.0       521.283432  2008  
51               88.0       1.365692               89.0       355.080000  2010  
31               84.0       8.744706               93.0       148.660000  2011  
4                84.0       0.652603               54.0        29.367143  2007  
35               84.0       0.825800               83.0         8.258000  2011  
21               83.0       6.636402               91.0       219.001261  2007  
61               82.0      10.180027               49.0       376.661000  2008  
43               82.0       5.103117               21.0       153.093505  2007  
53               81.0       1.783944               73.0       285.431000  2008  
1                81.0       8.096000               87.0        60.720000  2009  
47               81.0       7.221796               49.0       415.253258  2008  
5                80.0       4.471875               84.0        14.310000  2011  
7                80.0       4.005737               93.0       340.487652  2007  
59               78.0      14.196400               27.0       709.820000  2009

In [37]:
fig = px.bar(audience_score_top_15, x='Film', y= 'Audience  score %',  hover_data = audience_score_top_15[['Year','Genre']], color='Genre')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

- Both ratings rated Wall E with the highest score.
- On the other hand, 'Twilight', 'PS I Love You' and 'The Twilight Saga' has lower score in Rotten Tomatoes than Audience Scores.

- And finally, let's see the most profitable  15 movies

#### Top 15 Most Profitable Movies

In [38]:
top_15_profit = df.sort_values('Profitability', ascending=False)[:15]
top_15_profit

Film    Genre   Lead Studio  \
8                            Fireproof    Drama   Independent   
15  High School Musical 3: Senior Year   Comedy        Disney   
59         The Twilight Saga: New Moon    Drama        Summit   
66                            Waitress  Romance   Independent   
61                            Twilight  Romance        Summit   
29                          Mamma Mia!   Comedy     Universal   
31                   Midnight in Paris  Romance          Sony   
1                 (500) Days of Summer   Comedy           Fox   
57                        The Proposal   Comedy        Disney   
47                    Sex and the City   Comedy  Warner Bros.   
14         He's Just Not That Into You   Comedy  Warner Bros.   
21                          Knocked Up   Comedy     Universal   
62             Twilight: Breaking Dawn  Romance   Independent   
69               What Happens in Vegas   Comedy           Fox   
60                      The Ugly Truth   Comedy   Independent   

    Audience  score %  Profitability  Rotten Tomatoes %  Worldwide Gross  Year  
8                51.0      66.934000               40.0        33.467000  2008  
15               76.0      22.913136               65.0       252.044501  2008  
59               78.0      14.196400               27.0       709.820000  2009  
66               67.0      11.089742               89.0        22.179483  2007  
61               82.0      10.180027               49.0       376.661000  2008  
29               76.0       9.234454               53.0       609.473955  2008  
31               84.0       8.744706               93.0       148.660000  2011  
1                81.0       8.096000               87.0        60.720000  2009  
57               74.0       7.867500               43.0       314.700000  2009  
47               81.0       7.221796               49.0       415.253258  2008  
14               60.0       7.153600               42.0       178.840000  2009  
21               83.0       6.636402               91.0       219.001261  2007  
62               68.0       6.383364               26.0       702.170000  2011  
69               72.0       6.267647               28.0       219.367646  2008  
60               68.0       5.402632               14.0       205.300000  2009

In [39]:
fig= px.bar(top_15_profit, x='Film', y= 'Profitability',  hover_data = top_15_profit[['Year','Genre']], color='Lead Studio')
fig.update_layout(xaxis={'categoryorder':'total descending'})
fig.show()

- Interestingly, or not interestingly, 'Fireproof', the most profitable movie in our dataset, has low score in both of the ratings.

- Almost all of the lead studios have their movies in the most profitable top 15 movies list.

## This notebook is a part of the 9 Beginner Friendly EDAs
## If you like this one, you can also check out other notebooks in the Beginner Friendly EDAs series!

* [Data Analyst Jobs - EDA](https://www.kaggle.com/kaanboke/plotly-data-analyst-jobs)
* [Top Games on Google Play Store](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-top-games)
* [UDEMY Courses EDA](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-udemy)
* [World Happiness Report - EDA](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-eda)
* [Countries Life Expectancy](https://www.kaggle.com/kaanboke/plotly-beginner-friendly)
* [Netflix Movies- EDA](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-netflix)
* [Amazon Top 50 Bestselling Books EDA](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-amazon)
* [London Bike Sharing EDA](https://www.kaggle.com/kaanboke/plotly-beginner-friendly-london-bike)


- Thanks for the dataset contibutor for this data. I really enjoyed working on it.

- It was a quite pleasure to share with you this detailed, beginner friendly EDA. Thanks for your time.

- All the best 